.github/workflows/publish.yml

```diff
@@ -0,0 +1,22 @@
+name: 'Publish to npm'
+on:
+  push:
+    tags:
+       - "v*.*.*"
```

Comment:
Docs: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#running-your-workflow-only-when-a-push-of-specific-tags-occurs